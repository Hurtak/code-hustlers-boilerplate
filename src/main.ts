import { zValidator } from "npm:@hono/zod-validator";
import { Hono } from "npm:hono";
import { logger } from "npm:hono/logger";
import { z } from "npm:zod";

import { config } from "./config.ts";

const app = new Hono();
app.use(logger());

app.get("/health-check", (c) => {
  return c.text("OK");
});

app.post(
  "/hello",
  zValidator(
    "query",
    z.object({
      name: z.string(),
    }),
  ),
  (c) => {
    const { name } = c.req.valid("query");

    return c.json({
      message: `Hello ${name}!`,
    });
  },
);

Deno.serve({ port: config.PORT }, app.fetch);
