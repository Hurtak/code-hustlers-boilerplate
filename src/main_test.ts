import { add } from "./main.ts";
import { assertEquals } from "@std/assert";
import { assertSnapshot } from "@std/testing/snapshot";
import { describe, test } from "@std/testing/bdd";

Deno.test("test 1", () => {
  assertEquals(add(1, 2), 3);
});

describe("add", () => {
  test("result", () => {
    assertEquals(add(1, 2), 3);
  });

  test("object", () => {
    assertEquals({ a: 1 }, { a: 1 });
  });
});

test("snapshot", async (t) => {
  await assertSnapshot(t, add(1, 2));
});
