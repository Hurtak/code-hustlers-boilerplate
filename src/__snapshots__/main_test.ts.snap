export const snapshot = {};

snapshot[`snapshot 1`] = `3`;
